VotingApplication/VotingApplication.Web.Api.Tests/E2E/BasicPollTests.cs

```diff
@@ -0,0 +1,603 @@
+
+                IWebElement formDescription = _driver.FindElement(NgBy.Model("addOptionForm.description"));
+                Assert.IsTrue(formDescription.IsVisible());
+                Assert.AreEqual(String.Empty, formDescription.Text);
+            }
+
+            [TestMethod, TestCategory("E2E")]
+            public void BasicVote_OptionAddingPrompt_AcceptsValidName()
+            {
+                _driver.Navigate().GoToUrl(SiteBaseUri + "Poll/#/Vote/" + _optionAddingBasicPoll.UUID);
+                IReadOnlyCollection<IWebElement> anchors = _driver.FindElements(By.TagName("a"));
+                IWebElement addOptionLink = anchors.First(a => a.Text == "+ Add another option");
+                addOptionLink.Click();
+
+                IWebElement formName = _driver.FindElement(NgBy.Model("addOptionForm.name"));
+
+                IReadOnlyCollection<IWebElement> buttons = _driver.FindElements(By.TagName("Button"));
+                IWebElement doneButton = buttons.First(b => b.Text == "Done");
+
+                Assert.IsTrue(doneButton.IsVisible());
+                Assert.IsFalse(doneButton.Enabled);
+
+                formName.SendKeys("New Option");
+
+                Assert.IsTrue(doneButton.Enabled);
+            }
+
+            [TestMethod, TestCategory("E2E")]
+            public void BasicVote_OptionAddingSubmission_AddsOption()
+            {
+                _driver.Navigate().GoToUrl(SiteBaseUri + "Poll/#/Vote/" + _optionAddingBasicPoll.UUID);
+                IReadOnlyCollection<IWebElement> anchors = _driver.FindElements(By.TagName("a"));
+                IWebElement addOptionLink = anchors.First(a => a.Text == "+ Add another option");
+                addOptionLink.Click();
+
+                IWebElement formName = _driver.FindElement(NgBy.Model("addOptionForm.name"));
+
+                IReadOnlyCollection<IWebElement> buttons = _driver.FindElements(By.TagName("Button"));
+                IWebElement doneButton = buttons.First(b => b.Text == "Done");
+
+                String newOptionName = "New Option";
+                formName.SendKeys(newOptionName);
+
+                IWebElement form = _driver.FindElement(By.Name("addOptionForm"));
+                form.Submit();
+
+                IReadOnlyCollection<IWebElement> optionNames = _driver.FindElements(NgBy.Binding("option.Name"));
+
+                Assert.AreEqual(_optionAddingBasicPoll.Options.Count + 1, optionNames.Count);
+                Assert.AreEqual(newOptionName, optionNames.Last().Text);
+
+                // Refresh to ensure they new option was stored in DB
+                _driver.Navigate().GoToUrl(SiteBaseUri + "Poll/#/Vote/" + _optionAddingBasicPoll.UUID);
+
+                optionNames = _driver.FindElements(NgBy.Binding("option.Name"));
+
+                Assert.AreEqual(_optionAddingBasicPoll.Options.Count + 1, optionNames.Count);
+                Assert.AreEqual(newOptionName, optionNames.Last().Text);
```

Comment:
Other tests call VoteClearer and remove the last vote. This test adds an option and checks it's saved, but never removes it?